pkg/services/ngalert/eval/extract_md.go

```diff
@@ -65,3 +65,24 @@ func extractEvalString(frame *data.Frame) (s string) {
 
 	return ""
 }
+
+// extractValues returns the RefID and value for all reduce and math expressions
+// in the frame. It does not return values for classic conditions as the values
+// in classic conditions do not have a RefID. It returns nil if there are
+// no results in the frame.
+func extractValues(frame *data.Frame) map[string]NumberValueCapture {
+	if frame == nil {
+		return nil
```

Comment:
Is `nil` the same as an empty map for all cases? I'm not very familiar with how go templates work but if we ended up doing `values.A` in the template wouldn't this panic in this case?

Reply:
I thought it would panic, but I can't seem to get it to https://play.golang.org/p/cQeGYhhHesL

Reply:
TIL

Reply:
The check for that is here https://github.com/golang/go/blob/master/src/text/template/exec.go#L976